Diagnostics/HealthChecker/DataCollection/ServerInformation/Get-ServerOperatingSystemVersion.ps1

```diff
@@ -1,39 +1,59 @@
 ﻿# Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
-. $PSScriptRoot\..\..\..\..\Shared\Get-WmiObjectHandler.ps1
+. $PSScriptRoot\..\..\..\..\Shared\Get-RemoteRegistryValue.ps1
 function Get-ServerOperatingSystemVersion {
     [CmdletBinding()]
-    [OutputType("System.String")]
+    [OutputType("System.Object")]
     param(
-        [string]$OsCaption
+        [string]$ComputerName = $env:COMPUTERNAME,
+
+        [ScriptBlock]$CatchActionFunction
     )
     begin {
         Write-Verbose "Calling: $($MyInvocation.MyCommand)"
         $osReturnValue = [string]::Empty
     }
     process {
-        if ([string]::IsNullOrEmpty($OsCaption)) {
-            Write-Verbose "Getting the local machine version build number"
-            $OsCaption = (Get-WmiObjectHandler -Class "Win32_OperatingSystem").Caption
+        Write-Verbose "Getting the version build information for computer: $ComputerName"
+        $baseParams = @{
+            MachineName         = $ComputerName
+            CatchActionFunction = $CatchActionFunction
+        }
+
+        # Get ProductName via registry call as this is more accurate when running on Server Core
+        $productNameParams = $baseParams + @{
+            SubKey   = "SOFTWARE\Microsoft\Windows NT\CurrentVersion"
+            GetValue = "ProductName"
         }
-        Write-Verbose "OsCaption: '$OsCaption'"
 
-        switch -Wildcard ($OsCaption) {
+        # Find out if we're running on Server Core to output on the 'Operating System Information' page
+        $installationTypeParams = $baseParams + @{
+            SubKey   = "SOFTWARE\Microsoft\Windows NT\CurrentVersion"
+            GetValue = "InstallationType"
+        }
```

Comment:
Doesn't matter to much, but for variables like this that aren't going to change, look better in `begin` block so that doesn't look so empty. Not a required change, just for future reference.

Reply:
Indeed. Will move it to begin